python/tk_multi_reviewsubmission/renderer.py

```diff
@@ -103,15 +114,149 @@ def gather_nuke_render_info(self, path, output_path,
+                    log.critical(msg)
+
+            # TODO: do we handle this differently?
+            # Replace the client number
+            elif var_tmp == 'clientnum':
+                if not hasattr(self, "clientnum"):
+                    attr = attr.replace(var, str(data.get('clientnum')))
+
+            # TODO: we handle this differently
+            elif var_tmp == 'version_data':
+                if not hasattr(self, "version_data"):
+                    attr = attr.replace(var, str(data.get('version_data')))
+
+            # TODO: do we handle this differently?
+            # Replace the lens variable
+            elif var_tmp == "lens":
+                if not hasattr(self, "slate_lens"):
+                    attr = attr.replace(var, str(data.get('slate_lens')))
+
+            # TODO: do we handle this differently?
+            # Replace the slate_notes, notes, daily_status variable
+            elif var_tmp == "slate_notes" or var_tmp == "notes" or var_tmp == "daily_status":
+                replaceval = str(data.get(var_tmp))
+                # escape characters
+                replaceval = replaceval.replace("\\", "\\\\")
+                replaceval = replaceval.replace("\"", "\\\"")
+                replaceval = replaceval.replace("\'", "\\\'")
+                attr = attr.replace(var, str(replaceval))
+
+            # remove knobs that have a [**] value but nothing in data
+            elif (data.get(var_tmp) == '' or
+                          data.get(var_tmp) == None or
+                          data.get(var_tmp) == "None"):
+                regexp = ".*\[\*"
+                regexp += var_tmp
+                regexp += "\*\].*"
+                line = re.compile(regexp).findall(attr)
+                if line != None and len(line) > 0:
+                    if line[0].count('message') > 0:
+                        attr = attr.replace(var, str('""'))
+                    else:
+                        attr = attr.replace(var, "None")
+
+            else:
+                replaceval = str(data.get(var_tmp))
+                if (replaceval == ""):
+                    replaceval == '""'
+                attr = attr.replace(var, str(replaceval))
+        return attr
+    # end replaceVars
+
     def render_movie_in_nuke(self, path, output_path,
                              width, height,
                              first_frame, last_frame,
                              version, name,
                              color_space,
+                             replace_data,
                              active_progress_info=None):
+        # replace tokens in self._burnin_nk and save it in /var/tmp
+        tmp_file_prefix = "reviewsubmission_tmp_nuke_script"
+        tmp_file_name = "%s_%s.nk" % (tmp_file_prefix, hashlib.md5(str(time.time())).hexdigest())
+        tmp_full_path = os.path.join("/var", "tmp", tmp_file_name)
+
+        # TESTING:
+        print "Saving to: ", tmp_full_path
```

Comment:
Debug print? If not, should use logger right?

Reply:
Changed to log_debug()